php-checks/src/test/resources/checks/UnusedPrivateMethodCheck.php

```diff
@@ -30,7 +30,7 @@ private function j() {            // OK - used as encapsulated variable in strin
   public function k() {             // OK - public
   }
 
-  public function __i() {           // OK - magic method
+  private function __i() {           // OK - magic method
```

Comment:
I'm a bit confused by this test :thinking: OTOH, docs claim that `All magic methods _must_ be declared public`, on the other - the actual `isMagicMethod` logic was never reached in the tests

Reply:
I don't get it either :sweat_smile: Looks like a mistake from the commit introducing this test as public.
From the documentation there is few exception to this conditions for [__construct()](https://www.php.net/manual/en/language.oop5.decon.php#object.construct), [__destruct()](https://www.php.net/manual/en/language.oop5.decon.php#object.destruct), and [__clone()](https://www.php.net/manual/en/language.oop5.cloning.php#object.clone).
Maybe in this test you can replace `__i` with `__clone`, to add consistancy.